### PR TITLE
#1776 adding a message when no qualifiers are found

### PR DESCRIPTION
**Pull request by sjaakd**

### CODE REVIEW

**Comment by sjaakd** [comment]
@filiphr / @chris922 .. does one of you guys have a possibility to review this one?

File: processor/src/main/java/org/mapstruct/ap/internal/model/MappingBuilderContext.java [hunk 1]
```diff
@@ -93,7 +93,13 @@
         Assignment getTargetAssignment(Method mappingMethod, Type targetType,
                                        FormattingParameters formattingParameters,
                                        SelectionCriteria criteria, SourceRHS sourceRHS,
-                                       AnnotationMirror positionHint);
+                                       AnnotationMirror positionHint,
+                                       Forger forger);
+
+        @FunctionalInterface
+        interface Forger {
```
Comment (filiphr) [inline processor/src/main/java/org/mapstruct/ap/internal/model/MappingBuilderContext.java]: Why not use `java.util.function.Supplier`? Also if we go with our own, maybe `AssignmentForger`?

Comment (sjaakd) [reply processor/src/main/java/org/mapstruct/ap/internal/model/MappingBuilderContext.java]: I went for the `Supplier`.. Perhaps better than inventing our own types..

**Comment by sjaakd** [comment]
@filiphr .. thanks for reviewing!

**Pull request merged**